### PR TITLE
issue 2580

### PR DESCRIPTION
**Pull request by edewit**
- added missing translations
- small wait in between load and delete query

fixes: #2580

### CODE REVIEW

File: src/components/role-mapping/RoleMapping.tsx [hunk 1]
```diff
@@ -172,12 +174,16 @@ export const RoleMapping = ({
   const [hide, setHide] = useState(false);
   const [showAssign, setShowAssign] = useState(false);
   const [selected, setSelected] = useState<Row[]>([]);
+  const [wait, toggle] = useToggle();
```
Comment (jonkoops) [inline src/components/role-mapping/RoleMapping.tsx:177]: Might be worth it to add what is being toggled to the name.
```suggestion
  const [wait, toggleWait] = useToggle();
```

File: src/components/role-mapping/RoleMapping.tsx [hunk 2]
```diff
@@ -172,12 +174,16 @@ export const RoleMapping = ({
...
+  const setTimeout = useSetTimeout();
 
   const assignRoles = async (rows: Row[]) => {
     await save(rows);
     refresh();
   };
 
+  useEffect(() => setTimeout(refresh, 200), [wait]);
```
Comment (jonkoops) [inline src/components/role-mapping/RoleMapping.tsx:185]: Interesting, why do you need to do this on a timeout?

**Review by jonkoops — APPROVED** [review]

File: src/components/role-mapping/RoleMapping.tsx (hunk 2 above)
Comment (edewit) [reply src/components/role-mapping/RoleMapping.tsx:185]: somehow it gives me the old values

Comment (jonkoops) [reply src/components/role-mapping/RoleMapping.tsx:185]: Strange, I guess it still takes a while before the transaction is done.

**Pull request merged**